base/abstractarray.jl

```diff
@@ -2276,7 +2283,7 @@ function _typed_hvncat(::Type{T}, dims::Tuple{Vararg{Int, N}}, row_first::Bool,
     len == outlen || ArgumentError("too many elements in arguments; expected $(outlen), got $(len)") |> throw
 
     # copy into final array
-    A = Array{T, nd}(undef, outdims...)
+    A = cat_similar(as[1], T, outdims)
```

Comment:
Would this make it type stable?
```suggestion
    A = cat_similar(as[1], T, NTuple{nd, Int}(outdims))
```
I think then the vararg and `Vector` versions of `cat_similar` could also be deleted, right?

Reply:
Unfortunately not:
```
const b = fill(1);
@btime [b b ;;; 1 1]
 627.811 ns (21 allocations: 1.17 KiB) # current
 2.433 μs (28 allocations: 1.39 KiB) # NTuplized
 795.556 ns (23 allocations: 1.23 KiB) # splatted
```
It looks like I can avoid the vararg version so, I can cut those methods out at least